modules/core/src/lib/parse.js

```diff
@@ -58,19 +57,21 @@ async function parseWithLoader(loader, data, options, context) {
     return loader.parseTextSync(data, options, context, loader);
   }
 
+  // If we have a worker and loader hasn't
+  if (options.worker) {
+    // TODO - move to workerUrl only.
+    if ((loader.worker || loader.workerUrl) && loader.useWorker(options)) {
```

Comment:
- `loader.workerUrl` or `options.workerUrl`?
- Where do we ensure that `loader.useWorker` is a function?

Reply:
Updated.